react/ShelfContent.js

```diff
@@ -1,172 +1,159 @@
-        {
-          breakpoint: BREAKPOINT_MOBILE_VIEWPORT,
-          settings: {
-            slidesToShow: SLIDES_TO_SHOW_MOBILE_VIEWPORT,
-            slidesToScroll: SLIDES_TO_SCROLL_MOBILE_VIEWPORT,
-            arrows: ARROWS_MOBILE_VIEWPORT,
-            dots: DOTS_MOBILE_VIEWPORT,
-            centerMode: SLIDER_CENTER_MOBILE_MODE,
-            variableWidth: VARIABLE_WIDTH_MOBILE_MODE,
-          },
-        },
-        {
-          breakpoint: BREAKPOINT_EXTRA_SMALL_MOBILE_VIEWPORT,
-          settings: {
-            slidesToShow: SLIDES_TO_SHOW_MOBILE_VIEWPORT,
-            slidesToScroll: SLIDES_TO_SCROLL_MOBILE_VIEWPORT,
-            arrows: ARROWS_MOBILE_VIEWPORT,
-            dots: DOTS_EXTRA_SMALL_MOBILE_VIEWPORT,
-            centerMode: SLIDER_CENTER_MODE_EXTRA_SMALL_MOBILE,
-            variableWidth: VARIABLE_WIDTH_MOBILE_MODE,
-          },
-        },
-      ],
-    }
+  state = {
+    currentSlide: 0,
   }
 
-  getClassByItemsPerPage(itemsPerPage) {
-    switch (itemsPerPage) {
-      case 5:
-        return 'w-20'
-      case 4:
-        return 'w-25'
-      case 3:
-        return 'w-third'
-    }
+  handleChangeSlide = i => {
+    this.setState({ currentSlide: i })
   }
 
-  get itemsToShow() {
-    const { itemsPerPage, width } = this.props
-    const maxItems = Math.floor(width / DEFAULT_SHELF_ITEM_WIDTH)
-    return maxItems <= itemsPerPage ? maxItems : itemsPerPage
+  handleNextSlide = () => {
+    this.setState(({ currentSlide }) => {
+      const bannersLength = this.props.banners.filter(
+        banner => banner && (banner.mobileImage || banner.image)
+      ).length
+      const nextSlide = (currentSlide + 1) % bannersLength
+
+      return {
+        currentSlide: nextSlide,
+      }
+    })
   }
 
-  get sliderWidth() {
-    const { width } = this.props
-
-    const items = this.itemsToShow
-    const slider = items * DEFAULT_SHELF_ITEM_WIDTH
+  arrowRender = ({ orientation, onClick }) => {
+    const containerClasses = classNames(
+      shelf.arrow,
+      'pointer z-1 flex absolute',
+      {
+        [shelf.arrowLeft]: orientation === 'left',
+        [shelf.arrowRight]: orientation === 'right',
+      }
+    )
+    return (
+      <div className={containerClasses} onClick={onClick}>
+        <IconCaret orientation={orientation} thin size={25} />
+      </div>
+    )
+  }
 
-    if (items >= ITEMS_TO_FULL_WIDTH || width <= slider) return width
+  ArrowContainerRender = ({ children }) => {
+    const wrapperClasses = classNames(
+      shelf.arrowsContainerWrapper,
+      'w-100 h-100 absolute left-0 top-0 flex justify-center'
+    )
+    const containerClasses = classNames(
+      shelf.arrowsContainer,
+      'w-100 h-100 mw9 flex-ns justify-between items-center dn-s'
+    )
 
-    return slider
+    return (
+      <div className={wrapperClasses}>
+        <Container className={containerClasses}>{children}</Container>
+      </div>
+    )
   }
 
   slideFallback = (item = {}, key, fullWidth) => {
```

Comment:
I don't think this function is needed any more, you can just create this style object inside the render method and pass it